### PR TITLE
Move profile controller spec to requests spec and refactor

### LINKED ISSUE #1834: Refactor spec/controllers/profile_controller_spec.rb to spec/request/profile_spec.rb
# Description

Controller Specs had been obsolete since Rails 5. It is best practice to create request spec instead of controller spec as recommended by the Rspec team.

Recommended Reading:

- [Stackoverflow Discussion in regards to Controller Specs vs Request Specs](https://stackoverflow.com/questions/40851705/controller-specs-vs-request-specs)
- [Medium article discussing request and controller specs](https://medium.com/just-tech/rspec-controller-or-request-specs-d93ef563ef11)

Resources on Request Spec formatting:

- [RelishApp](https://relishapp.com/rspec/rspec-rails/v/4-0/docs/request-specs/request-spec)

Original Issue: https://github.com/ifmeorg/ifme/issues/1815

---

Please assign yourself (via the Assignees dropdown), if you do want to work on this issue. Can't find yourself? You need to [join our organization](https://github.com/ifmeorg/ifme/wiki/Join-Our-Slack).

Check out our [Picking Up Issues](https://github.com/ifmeorg/ifme/wiki/Picking-Up-Issues) guide if you haven't already!

### PR DESCRIPTION
**Pull request by DRBragg**
# Description

This PR refactors the tests in `profile_controller_spec.rb` to use RSpec request tests instead of controller tests. It replace PR #1882 as there was an unresolvable issue with the marge/rebase of the main brach in the PR

## More Details

Confirmed coverage of ProfileController remained at 100% after refactor.

## Corresponding Issue

Closes #1834 

---

Reviewing this pull request? Check out our [Code Review Practices](https://github.com/ifmeorg/ifme/wiki/Code-Review-Practices) guide if you haven't already!

### CODE REVIEW

**Comment by DRBragg** [comment]
@bennpham I wasn't able to figure out how to fix the merge/rebase issue in #1882 so I move all the changes into this PR. Can you confirm that all is good with this PR and I'll close the other one?

Sorry for the delay getting this all wrapped up. I got a concussion at hockey and I'm just getting back on to the computer.

**Comment by bennpham** [comment]
@DRBragg Ouch. Sorry to hear about that! Hope you recover well. Yeah, I have some crazy events going around here and there on my end as well and some good news that pop up every now and then throughout that process so I'll try to check on some things when I can then go back to tend to those situations.

File: spec/requests/profile_spec.rb [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+          expect(flash[:notice]).to eq("Ban removed on #{banned_user.name}")
+        end
+      end
+
+      context "when user does not exist" do
+        it "does not remove ban" do
+          expect { post remove_ban_profile_index_path, params: {user_id: -1} }
+            .not_to change(Devise.mailer.deliveries, :count).from(0)
+          expect(response).to redirect_to(admin_dashboard_path)
+          expect(flash[:alert]).to eq("Could not remove ban on -1")
```
Comment (bennpham) [inline spec/requests/profile_spec.rb:159]:
```suggestion
          expect(flash[:alert]).to include("Could not remove ban")
```

Nitpick on all the flash alert equals, but you can also use `include` and just put part of the string as well just to make the less likely to break if its a simple verbiage change on the string. Overall, it's not a big deal. I consider the tests ok.

**Review by bennpham — APPROVED** [review]
Good job. I approved. I have one nitpick comment that's just a suggestion but your approach is fine as well 👍

**Comment by bennpham** [comment]
Can you also close out https://github.com/ifmeorg/ifme/pull/1882 ?

File: spec/requests/profile_spec.rb (hunk 1 above)
Comment (DRBragg) [reply spec/requests/profile_spec.rb:159]: That works for me.

**Comment by DRBragg** [comment]
Yup! done.

**Review by julianguyen — APPROVED** [review]
Go ahead and merge when you're ready! Thanks for taking this on :)

**Pull request merged**